packages/mds-logger/index.ts

```diff
@@ -14,231 +14,26 @@
-
-import { WebClient as SlackClient } from '@slack/client'
-
-const { env } = process
-
-type LogLevel = 'INFO' | 'WARN' | 'ERROR'
-
-interface Datum {
-  lat?: string
-  lng?: string
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  [propName: string]: any
-}
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-let pushClient: any
-
-if (env.PUSHOVER_TOKEN) {
-  pushClient = new PushClient({
-    user: env.PUSHOVER_USER,
-    token: env.PUSHOVER_TOKEN,
-    debug: true
-  })
-}
-
-async function sendPush(msg: string, priority?: number) {
-  if (!pushClient) {
-    return
-  }
-  const payload = {
-    // These values correspond to the parameters detailed on https://pushover.net/api
-    // 'message' is required. All other values are optional.
-    message: msg, // required
-    // title: 'Test Pushover',
-    // sound: 'magic',
-    // device: 'devicename',
-    priority: priority || 0
-  }
-
-  const [err, result] = await pushClient.send(payload)
-  if (err) {
-    console.error('ERROR pushover fail', err)
-    throw err
-  } else {
-    console.log('INFO pushover success', result)
-    return result
-  }
-}
-
-function makeCensoredDatum(datum: Datum) {
-  if (datum instanceof Error) {
-    return datum.toString()
-  }
-  const censoredDatum: Datum = { ...datum }
-  if (censoredDatum.lat != null) {
-    censoredDatum.lat = 'CENSORED_LAT'
-  }
-  if (censoredDatum.lng != null) {
-    censoredDatum.lng = 'CENSORED_LNG'
-  }
-  return censoredDatum
-}
-
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-function isAtomic(item: any) {
-  const type = typeof item
-  return ['string', 'number', 'boolean', 'undefined'].includes(type) || item == null
-}
-
-// just in case we have to censor something nested like
-// { data: [{ lat:1, lng:2 }] }
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-function makeCensoredLogMsgRecurse(msg: any): any {
-  if (isAtomic(msg)) {
-    return msg
-  }
-  if (Array.isArray(msg)) {
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    return msg.map((arr_item: any) => makeCensoredLogMsgRecurse(arr_item))
-  }
-  const censoredObject = makeCensoredDatum(msg)
-  if (typeof censoredObject === 'object') {
-    Object.keys(censoredObject).forEach((key: string) => {
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      const val: any = censoredObject[key]
-      if (!isAtomic(val)) {
-        censoredObject[key] = makeCensoredLogMsgRecurse(val)
-      }
-    })
-  }
-  return censoredObject
-}
-
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-function makeCensoredLogMsg(...msgs: any[]): any[] {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  return makeCensoredLogMsgRecurse(msgs).map((item: any) =>
-    // never print out '[object Object]'
-    String(item) === '[object Object]' ? JSON.stringify(item) : item
+const redact = (args: unknown[]): string[] =>
```

Comment:
Do we not care about the redaction working on arbitrarily nested objects? This, for instance, fails if an array of objects containing GPS information is logged.

Reply:
Why would you say it fails? Did you try it? All tests pass, plus:

```ts
logger.info([
  { gps: { lat: 0, lng: 0 } },
  {
    points: [
      { lat: 0, lng: 0 },
      { lat: 1, lng: 1 },
      { lat: 2, lng: 2 }
    ]
  }
])
```
Produces:
`INFO [{"gps":{"lat":"[REDACTED]","lng":"[REDACTED]"}},{"points":[{"lat":"[REDACTED]","lng":"[REDACTED]"},{"lat":"[REDACTED]","lng":"[REDACTED]"},{"lat":"[REDACTED]","lng":"[REDACTED]"}]}]`

Reply:
This does not fail in that case:
```
INFO [{"device_id":"ec551174-f324-4251-bfed-28d9f3f473fc","gps":{"lat":"[REDACTED]","lng":"[REDACTED]","speed":0,"hdop":1,"heading":180},"charge":0.5,"timestamp":1555384091559,"recorded":1555384091836},{"device_id":"ec551174-f324-4251-bfed-28d9f3f473fc","gps":{"lat":"[REDACTED]","lng":"[REDACTED]","speed":100,"hdop":10,"heading":20},"charge":0.75,"timestamp":1555384090000,"recorded":1555384090000}]
```

Reply:
> Why would you say it fails? Did you try it? All tests pass, plus:
> 
> ```ts
> logger.info(
>   { gps: { lat: 0, lng: 0 } },
>   {
>     points: [
>       { lat: 0, lng: 0 },
>       { lat: 1, lng: 1 },
>       { lat: 2, lng: 2 },
>       {
>         more: [
>           { lat: 3, lng: 3 },
>           { lat: 4, lng: 4 }
>         ]
>       }
>     ]
>   }
> )
> ```
> 
> Produces:
> `INFO {"gps":{"lat":"[REDACTED]","lng":"[REDACTED]"}} {"points":[{"lat":"[REDACTED]","lng":"[REDACTED]"},{"lat":"[REDACTED]","lng":"[REDACTED]"},{"lat":"[REDACTED]","lng":"[REDACTED]"},{"more":[{"lat":"[REDACTED]","lng":"[REDACTED]"},{"lat":"[REDACTED]","lng":"[REDACTED]"}]}]}`

I tried passing in an array to one of the tests and saw a failure, but that was my mistake. You're right, it does work.